apidef/validator.go

```diff
@@ -99,10 +100,18 @@ var ErrAllAuthSourcesDisabled = "all auth sources are disabled for %s, at least
 type RuleAtLeastEnableOneAuthSource struct{}
 
 func (r *RuleAtLeastEnableOneAuthSource) Validate(apiDef *APIDefinition, validationResult *ValidationResult) {
-	for k, authConfig := range apiDef.AuthConfigs {
-		if shouldValidateAuthSource(k, apiDef) && !(authConfig.UseParam || authConfig.UseCookie || !authConfig.DisableHeader) {
+	var authConfigs []string
```

Comment:
preallocate with make()

Reply:
What value will it bring?

Reply:
Yes it may be more performant in case huge numbers. Here it is less than 10. However, I can do that as well. Let me change.